parsec/core/gui/main_window.py

```diff
@@ -512,106 +532,106 @@ def add_new_tab(self):
+                        self,
+                        _("TEXT_FILE_LINK_PLEASE_LOG_IN_organization").format(
+                            organization=file_link_addr.organization_id
+                        ),
+                    )
+                    break
+            else:
+                show_error(
+                    self,
+                    _("TEXT_FILE_LINK_NOT_IN_ORG_organization").format(
+                        organization=file_link_addr.organization_id
+                    ),
+                )
 
+    def go_to_file_link(self, addr: BackendOrganizationFileLinkAddr) -> None:
+        # Try to use the file link on the already logged in cores
         for idx in range(self.tab_center.count()):
             if self.tab_center.tabText(idx) == _("TEXT_TAB_TITLE_LOG_IN_SCREEN"):
                 continue
+
             w = self.tab_center.widget(idx)
             if (
                 not w
                 or not w.core
-                or w.core.device.organization_addr.organization_id != action_addr.organization_id
+                or w.core.device.organization_addr.organization_id != addr.organization_id
             ):
                 continue
-            user_manifest = w.core.user_fs.get_user_manifest()
-            found_workspace = False
-            for wk in user_manifest.workspaces:
-                if not wk.role:
-                    continue
-                if wk.id == action_addr.workspace_id:
-                    found_workspace = True
-                    central_widget = w.get_central_widget()
-                    try:
-                        central_widget.go_to_file_link(wk.id, action_addr.path)
-                        self.switch_to_tab(idx)
-                    except AttributeError:
-                        logger.exception("Central widget is not available")
-                    return
-            if not found_workspace:
+
+            central_widget = w.get_central_widget()
+            if not central_widget:
+                continue
+
+            try:
+                central_widget.go_to_file_link(addr)
+
+            except GoToFileLinkBadOrganizationIDError:
+                continue
+            except GoToFileLinkBadWorkspaceIDError:
                 show_error(
                     self,
                     _("TEXT_FILE_LINK_WORKSPACE_NOT_FOUND_organization").format(
-                        organization=action_addr.organization_id
+                        organization=addr.organization_id
                     ),
                 )
                 return
-        show_info(
-            self,
-            _("TEXT_FILE_LINK_PLEASE_LOG_IN_organization").format(
-                organization=action_addr.organization_id
-            ),
-        )
-        self.switch_to_login_tab(action_addr)
+            except GoToFileLinkPathDecryptionError:
+                show_error(self, _("TEXT_INVALID_URL"))
+                return
+            else:
+                self.switch_to_tab(idx)
```

Comment:
Isn't it confusing to switch tab only if processing the file link succeeds? That means the error is going to be prompted in an unrelated tab.

Reply:
yep, good catch !